### PR TITLE
Problem: missing field of networkFee in old database #322

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: add legacy support

### CODE REVIEW

**Review by crypto-matto — COMMENTED** [review]
螢幕截圖 2021-03-25 上午10 25 46

We have to change here as well. `settings.tsx`
```
const onFinish = async values => {
    if (
      defaultSettings.nodeUrl === values.nodeUrl &&
      defaultSettings.indexingUrl === values.indexingUrl &&
      defaultSettings.network.chainId === values.chainId &&
      defaultSettings.fee.gasLimit === values.gasLimit &&
      defaultSettings.fee.networkFee === values.networkFee
    ) {
      // No value was updated, we stop here
      return;
    }
```

File: src/pages/settings/settings.tsx [hunk 1]
```diff
@@ -107,27 +109,42 @@ const FormSettings = () => {
...
-        gasLimit: defaultSettings.fee.gasLimit,
+        networkFee,
+        gasLimit,
       });
     }
   }, [form, defaultSettings]);
 
   const onFinish = async values => {
+    const defaultGasLimit =
+      defaultSettings.fee !== undefined ? defaultSettings.fee.gasLimit : FIXED_DEFAULT_GAS_LIMIT;
```
Comment (linfeng-crypto) [inline src/pages/settings/settings.tsx:138]: may be we can use a function getFee and getGasLimit, and not put it into defaultSettings

Comment (leejw51crypto) [reply src/pages/settings/settings.tsx:138]: ok

Comment (leejw51crypto) [reply src/pages/settings/settings.tsx:138]: https://github.com/crypto-com/chain-desktop-wallet/issues/324

i made another issue

File: src/pages/settings/settings.tsx [hunk 2]
```diff
@@ -107,27 +109,42 @@ const FormSettings = () => {
...
   const history = useHistory();
+  let networkFee = parseInt(FIXED_DEFAULT_FEE, 10);
+  let gasLimit = parseInt(FIXED_DEFAULT_GAS_LIMIT, 10);
 
   useEffect(() => {
     if (!didMountRef.current) {
       didMountRef.current = true;
+
+      if (defaultSettings.fee !== undefined) {
+        networkFee = parseInt(defaultSettings.fee.networkFee, 10);
```
Comment (eddycpc) [inline src/pages/settings/settings.tsx:120]: This value might be bigger than int, I don't think the parseInt is necessary

**Review by eddycpc — CHANGES_REQUESTED** [review]

File: src/pages/settings/settings.tsx (hunk 2 above)
Comment (leejw51crypto) [reply src/pages/settings/settings.tsx:120]: ok, i'll change it

**Review by eddycpc — APPROVED** [review]
Looks great!

For people with old DBs, please confirm to us that this has now been resolved 🙇

**Comment by crypto-matto** [comment]
> Looks great!
> 
> For people with old DBs, please confirm to us that this has now been resolved 🙇

I tested with old DB. Working good now.

**Pull request merged**